File: 03/03.pl
while(<>) { chomp; $in.=$_ }
sub mulsum { my $sum=0;
    while($_[0] =~ /mul\((\d+),(\d+)\)/g) {
        $sum += $1*$2 } $sum }

print mulsum($in)."\n";
print mulsum($in =~ s/don't\(\).*?(do\(\)|$)//gr)."\n";

File: 03/03.py
import re
inp = open(0).read()

def mulsum(s):
    ms = re.findall(r"mul\((\d+),(\d+)\)", s)
    return sum(int(a)*int(b) for a,b in ms)

print(mulsum(inp))

inp2 = re.sub(r"don't\(\).*", "", re.sub(r"don't\(\).*?do\(\)", "", re.sub("\n", "", inp)))
print(mulsum(inp2))
